packages/clay-autocomplete/src/DropDown.tsx

```diff
@@ -0,0 +1,62 @@
+/**
+ * © 2019 Liferay, Inc. <https://liferay.com>
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+import ClayDropDown from '@clayui/drop-down';
+import Context from './Context';
+import React, {useContext} from 'react';
+
+export interface IProps extends React.HTMLAttributes<HTMLDivElement> {
+	/**
+	 * HTML element that the menu should be aligned to
+	 */
+	alignElementRef?: React.RefObject<HTMLElement>;
+
+	/**
+	 * Flag to indicate if menu is showing or not.
+	 */
+	active?: boolean;
+
+	/**
+	 * Callback function for when active state changes.
+	 */
+	onSetActive?: (val: boolean) => void;
+}
+
+const DropDown: React.FunctionComponent<IProps> = ({
+	active = false,
+	alignElementRef,
+	children,
+	onSetActive = () => {},
+}) => {
+	const {containerElementRef} = useContext(Context);
+
+	if (!alignElementRef) {
+		alignElementRef = containerElementRef;
+	}
+
+	const menuElementRef = React.useRef<HTMLDivElement>(null);
```

Comment:
For sake of readability, I'd probably try to keep hooks towards the top of the function declaration.

I'm open to other ideas, but this is generally what I see in other projects.

Reply:
Makes sense, I think it's worth adopting as a practice.